### PR TITLE
[MO] - [refactor] -> ClientViaHttp to ClientViaExec

### PR DESCRIPTION
**Pull request by Seequick1**
### Type of change

- Enhancement / new feature
- Refactoring

### Description

This is only draft PR. Refactor, which change how we exchange messages between kafka clients. It worked on logic when clients were created via HTTP request and now kafka clients are creating inside of the pod. 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Make sure all tests pass

### CODE REVIEW

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=regression testcase=KafkaST+MirrorMakerST+SecurityST

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=KafkaST,MirrorMakerST,SecurityST

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=regression testcase=KafkaST,MirrorMakerST,SecurityST

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=acceptance testcase=KafkaST,MirrorMakerST,SecurityST

**Comment by Seequick1** [comment]
# Failed because of failed because of Kafka pod creation common problem. 
- testLoadBalancerTls 
- testAutoRenewCaCertsTriggeredByAnno
# I tried this test on baremetal and it's ok
- testKafkaAndZookeeperScaleUpScaleDown

File: systemtest/src/main/java/io/strimzi/systemtest/MessagingBaseST.java [hunk 1]
```diff
@@ -31,107 +24,112 @@
...
-        response = cliApiClient.sendAndGetStatus(producer);
-
-        waitTillProcessFinish(getClientUUID(response), "producer", timeout);
-
-        assertThat(String.format("Return code of sender is not 0: %s", response),
-                response.getInteger("ecode"), is(0));
 
-        sent = getSentMessagesCount(response, messageCount);
+        boolean hasPassed = producer.run();
+        LOGGER.info("Producer -----> {}", hasPassed);
```
Comment (Frawless) [inline systemtest/src/main/java/io/strimzi/systemtest/MessagingBaseST.java]:
```suggestion
        LOGGER.info("Producer finished correctly: {}", hasPassed);
```

File: systemtest/src/main/java/io/strimzi/systemtest/MessagingBaseST.java [hunk 2]
```diff
@@ -152,44 +152,35 @@ int receiveMessages(int messageCount, long timeout, String clusterName, boolean
...
-        response = cliApiClient.sendAndGetStatus(consumer);
-
-        waitTillProcessFinish(getClientUUID(response), "consumer", timeout);
-
-        assertThat(String.format("Return code of receiver is not 0: %s", response),
-                response.getInteger("ecode"), is(0));
 
-        received = getReceivedMessagesCount(response);
+        boolean hasPassed = consumer.run();
+        LOGGER.info("Consumer -----> {}", hasPassed);
```
Comment (Frawless) [inline systemtest/src/main/java/io/strimzi/systemtest/MessagingBaseST.java]:
```suggestion
        LOGGER.info("Consumer finished correctly: {}", hasPassed);
```

File: docker-images/test-client/Dockerfile [hunk 3]
```diff
@@ -1,17 +1,9 @@
...
 
-ADD tmp/test-client-${STRIMZI_VERSION}.jar /test-client.jar
-
-USER 1001
-
-EXPOSE 4242/tcp
-
-CMD ["/opt/kafka/launch_java.sh", "/test-client.jar"]
\ No newline at end of file
+USER 1001
```
Comment (tombentley) [inline docker-images/test-client/Dockerfile]: It looks like you're pretty much deleting the test-client module, but for some reason you're keeping the `Dockerfile` and `Makefile`. Why?

Comment (Frawless) [reply docker-images/test-client/Dockerfile]: We removed `test-client` package, but `test-client` image is still needed for the tests with our "tooling" scripts for producers and consumers

**Pull request merged**